lambdas/data-migration2/src/executions.ts

```diff
@@ -0,0 +1,108 @@
+import Knex from 'knex';
+import Logger from '@cumulus/logger';
+
+import DynamoDbSearchQueue from '@cumulus/aws-client/DynamoDbSearchQueue';
+import { envUtils } from '@cumulus/common';
+import { ExecutionRecord } from '@cumulus/types/api/executions';
+import { PostgresExecutionRecord, translateApiExecutionToPostgresExecution } from '@cumulus/db';
+import { RecordAlreadyMigrated } from './errors';
+import { MigrationSummary } from './types';
+
+const Manager = require('@cumulus/api/models/base');
+const schemas = require('@cumulus/api/models/schemas');
+const Execution = require('@cumulus/api/models/executions');
+
+const logger = new Logger({ sender: '@cumulus/data-migration/executions' });
+
+/**
+ * Migrate execution record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Source record from DynamoDB
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<number>} - Cumulus ID for record
+ * @throws {RecordAlreadyMigrated}
+ *   if record was already migrated
+ */
+export const migrateExecutionRecord = async (
+  dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex
+): Promise<number> => {
+  // Use API model schema to validate record before processing
+  Manager.recordIsValid(dynamoRecord, schemas.execution);
+
+  const existingRecord = await knex<PostgresExecutionRecord>('executions')
+    .where({
+      arn: dynamoRecord.arn,
+    })
+    .first();
+  // Throw error if it was already migrated.
+  if (existingRecord) {
+    throw new RecordAlreadyMigrated(`Execution arn ${dynamoRecord.arn} was already migrated, skipping`);
+  }
+
+  const updatedRecord = await translateApiExecutionToPostgresExecution(
+    <ExecutionRecord>dynamoRecord, <Knex>knex
+  );
+
+  // If we have a parent ARN from the dynamo record but we couldn't find a cumulus_id in Postgres,
+  // we need to migrate the parent dynamo record to Postgres
+  if (dynamoRecord.parentArn !== undefined && updatedRecord.parent_cumulus_id === undefined) {
+    // Get parent record from Dynamo
+    const executionModel = new Execution();
+    const parentExecution = await executionModel.get({ arn: dynamoRecord.parentArn });
+
+    // Migrate parent dynamo record to Postgres and assign parent's cumulus_id to child
+    updatedRecord.parent_cumulus_id = await migrateExecutionRecord(parentExecution, knex);
+  }
+
+  const cumulusId = await knex('executions').insert(updatedRecord).returning('cumulus_id');
+
+  return cumulusId[0];
```

Comment:
```suggestion
  const [cumulusId] = await knex('executions').insert(updatedRecord).returning('cumulus_id');
  return cumulusId;
```

Reply:
Yup, that's neater. Or how about:

```
const cumulusId = await knex('executions').insert(updatedRecord).returning('cumulus_id').first();

return cumulusId;
```

?

Reply:
I don't think `.first()` does anything on an `insert` query. And you still have to destructure the column value out of the query result.

Reply:
Nope. Edit: This was a lie. `first()` doesn't work on inserts.